src/productList/ProductList.jsx

```diff
@@ -0,0 +1,157 @@
+import React, { useState, useEffect, useTransition } from 'react'
+import { Card, Input, Spin, Alert, AutoComplete } from 'antd'
+import { reqProductList, reqSearchProduct } from '../api'
+import './produc-list.less'
+const { Meta } = Card;
+const { Search } = Input;
+
+let cutArr = [] //用来存放切割后的数组
+let cutArrKey = 1
+
+export default function ProductList() {
+    const [productList, setProductList] = useState([])
+    const [isLoading, setIsLoading] = useState(true)
+    const [option, setOption] = useState([])
+    const [open, setOpen] = useState(false)
+    const [isPending, starTransition] = useTransition()
+
+    useEffect(() => {
+        const getProductList = async () => {
+            const result = await reqProductList()
+            const { products } = result.data
+            handleCutArr(products)
```

Comment:
You can set full list to the state, and get current list based on some state like `pageSize`, `currentPage`